### DISCUSSION TITLE
dispatcher process interrupted

### DISCUSSION THREAD

**Discussion by dmartinb06** [AbsaOSS/spline-spark-agent#840] [Q&A] [answered]
Hi everyone!

I have a spark application using the spline agent and a custom dispatcher. The spark application generates multiple plans and terminates before the dispatcher has finished processing all the plans, what do I have to do to make my spark application wait for the dispatcher to finish?

**Comment by cerveada** [comment]
Hello, what version of spark agent do you use?

**Reply by dmartinb06** [reply]
Hi Adam,

This is my dependency:

<dependency>
            <groupId>za.co.absa.spline.agent.spark</groupId>
            <artifactId>agent-core_2.12</artifactId>
            <version>1.1.0</version>
</dependency>

**Reply by cerveada** [reply]
Can you please try it with the latest agent version 2.2.0?
You could also try to set `hadoop.service.shutdown.timeout` to some high value.

If the problem is still there, please open an issue and add a log from the agent.

**Comment by dmartinb06** [comment, marked as answer]
Hello again,

in the end the problem was that the AsyncEventQueue was filling up because our dispatcher was taking too long, we solved it by increasing the parameters 
```spark.scheduler.listenerbus.eventqueue.capacity```
and 
```spark.scheduler.listenerbus.eventqueue.shared.capacity```

**Reply by cerveada** [reply]
Thanks for sharing the solution.
